### DISCUSSION TITLE
Allow for teacher customization of lessons

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#775] [Large Projects & Plans]
This is not necessarily a challenging idea, but I suspect a time consuming one that I will certainly not pick up myself. 

So what is the issue? In the current implementation, kids in classes see all adventures. But I can imagine, thinking as a teacher, that you would want to customize the adventures kids see. I imagine the following possibilities (in increasing order of complexity)

* Select from the set of adventures x levels the ones you want to show to students of a class.
* Override adventure texts with your own ideas/additions
* Create entirely new adventures for your class
* Commit these new adventures to the general adventures set so other teachers can use them too
* Set release dates on adventures per level so you can plan a whole lesson set in advance and do not have to open them each week

Similarly, teachers likely will want to be able to customize quizzes by:
* Selecting questions for a quiz
* Adding their own questions
* Commit these new questions to the general set so other teachers can use them too

**Comment by MarleenGilsing** [comment]
I really like the idea to give teachers the opportunity to costumize Hedy for their students. I think the most valuable feature would be selecting the adventures, because there are a lot of adventures at the moment and it would give the students a good oversight of what has to be done. 
I don't know if you've thought of it yet, but Hedy is probably going to be used by teachers with multiple different classes, so it would be nice for them to choose a class and plan some adventures for that class and maybe plan different adventures for a different class. It might also come in handy (mainly in primary education) to be able to plan different adventures for different students in the same class. For example I'd like to send the harder adventures to the stronger students and the easier ones to the students who struggle. 
Of course, a teacher could also just tell the students 'you start at this adventure and you start at this adventure'. 

I think the costumized test can be very valuable as well. Mainly adding your own questions. I think teachers like to have control over the content and difficulty of tests, so if they could add their own questions and remove questions they don't like that would be great. Maybe it would be nice too for them to give a certain set of questions a name and save it, so they could use the same settings for multiple classes, years or collegae could the same quiz too. 

Another cool feature, that would really suit the current climate in education, would be adaptive assessment. This means we'd have to make a bunch of different questions and rate them from easiest to hardest and monitor the students skills as well. And as the student takes the quiz (s)he gets harder or easier questions based on their performance on previous questions. I can't estimate how much work this is to program, a lot I reckon, so I don't know if it's worth the trouble, but it would be nice. (In this scenario teachers don't have any control over the quiz questions which would be a disadventage).

**Comment by Felienne** [comment]
@MarleenGilsing made a supercool mockup for this!!

![image](https://user-images.githubusercontent.com/1003685/136528733-7a346b37-50bb-4a75-9062-ee74e0f0cf57.png)

**Comment by Felienne** [comment]
With #1317 we are really close to this being finished!

The only thing we do not yet have is quizzes customization, but that cannot be done properly until the localization of quizzes is finished (will make a separate issue for that)
